src/lib.rs

```diff
@@ -80,6 +80,95 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+/// Wrapper for native `Dav1dSettings` struct.
+#[derive(Debug)]
+pub struct Settings {
+    dav1d_settings: Dav1dSettings,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Settings {
+    /// Creates a new [`Settings`] instance populated by `dav1d_default_settings`.
+    pub fn new() -> Self {
+        unsafe {
+            let mut dav1d_settings = mem::MaybeUninit::uninit();
+
+            dav1d_default_settings(dav1d_settings.as_mut_ptr());
+
+            Self {
+                dav1d_settings: dav1d_settings.assume_init(),
+            }
+        }
+    }
+
+    pub fn set_n_frame_threads(self: &mut Self, n_frame_threads: u32) {
+        self.dav1d_settings.n_tile_threads = n_frame_threads as i32;
```

Comment:
```suggestion
        self.dav1d_settings.n_frame_threads = n_frame_threads as i32;
```

Reply:
Swear I double-checked those :)

Reply:
The kind of mistakes you never notice in your own code :)